File: No_930/Binary Subarrays With Sum.cpp
class Solution {
public:
    int numSubarraysWithSum(vector<int>& nums, int goal) {
        unordered_map<int, int> freq;
        freq[0] = 1;
        
        int sum = 0, ret = 0;
        for(int i=0; i< nums.size(); i++) {
            sum += nums[i];
            ret += freq[sum-goal];
            freq[sum]++;
        }
        
        return ret;
    }
};

File: No_930/Binary Subarrays With Sum.py
class Solution:
    def numSubarraysWithSum(self, nums: List[int], goal: int) -> int:
        ret = 0
        prefix_sum = 0
        freq = defaultdict(int)
        freq[0] = 1
        
        for num in nums:
            prefix_sum += num
            ret += freq[prefix_sum-goal]
            freq[prefix_sum] += 1
    
        return ret